jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -180,12 +198,28 @@ def search(self, error_on_fail=False):
 
         # QTable stores all the filtered sky coordinate points to be marked
         catalog_results = QTable({'coord': filtered_skycoord_table})
-        self.number_of_results = len(catalog_results)
 
+        self.number_of_results = len(catalog_results)
         # markers are added to the viewer based on the table
         viewer.marker = {'color': 'red', 'alpha': 0.8, 'markersize': 5, 'fill': False}
         viewer.add_markers(table=catalog_results, use_skycoord=True, marker_name=self._marker_name)
 
+        # adding in coordinates and Source IDs into the catalog table
+        if self.catalog_selected == "SDSS":
+            for row in self.app._catalog_source_table:
+                row_info = {'Right Ascension (degrees)': row['ra'],
+                            'Declination (degrees)': row['dec'],
+                            'Object ID': row['objid']}
+                self.table.add_item(row_info)
+
+        if self.catalog_selected == 'From File...':
+            for row in self.app._catalog_source_table:
+                # find new to add in a way to append the source id to the table
+                # 'Object ID': row['label']} ; 'label' is failing tests
```

Comment:
instead of repeating these if-statements, can we move this logic into the previous if/elif statements and reference the object directly instead of having to rely on `self.app._catalog_source_table`?  

Or would `SkyCoordTable` which is populated for each case and then used for the markers be able to take the object IDs as well, in which case we wouldn't need separate cases at all and would also benefit from being able to act on `filtered_skycoord_table` in case any entries have been removed (otherwise we'll have to deal with this later when implementing click-to-zoom functionality since some entries might be missing marks)?

Reply:
Hey Kyle, I moved the code into the original if/elif statements. Could you further explain what you mean when you say "reference the object directly"? Are you referring to query_region_result?

Reply:
Awesome thanks!  I just meant that you could use `skycoord_table` and `table` instead of `self.app._catalog_source_table` since I'm guessing we'll need to get rid of that eventually, but that really doesn't matter and I can see arguments both ways - feel free to leave it as is now!